Libraries/LibGUI/GTextEditor.h

```diff
@@ -244,10 +248,11 @@ class GTextEditor
     RefPtr<GAction> m_delete_action;
     CElapsedTimer m_triple_click_timer;
     NonnullRefPtrVector<GAction> m_custom_context_menu_actions;
-    NonnullOwnPtrVector<NonnullOwnPtrVector<UndoCommand>> m_undo_stack;
-    int m_undo_index = 0;
+    NonnullOwnPtrVector<UndoCommandsContainer> m_undo_stack;
+    int m_undo_stack_size = 0;
+    int m_redo_stack_size = 0;
     int m_undo_timer = 0;
-    int m_prev_undo_stack_size = 0;
+    int m_prev_undo_vector_size = 0;
```

Comment:
This is not a great name, since it doesn't tell us anything about *when* "prev" refers to.
Can we come up with a better name that also encodes when the size snapshot was taken?